physionet-django/events/templates/events/event_home.html

```diff
@@ -11,193 +11,255 @@
-          </div>
+  {% for event in events_past %}
+  {% for info in event_details|get_applicant_info:event.id %}
+  <div class="modal fade" id="{{ info.id }}-modal-{{ event.id }}" tabindex="-1" role="dialog"
+    aria-labelledby="{{ info.id }}-modal" aria-hidden="true">
+    <div class="modal-dialog modal-lg" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title">{{ info.title }}</h5>
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <span aria-hidden="true">&times;</span>
+          </button>
         </div>
-    {% endfor %}
-    {% endfor %}
-  {% endif %}
+        {% include 'events/event_applications.html' %}
+      </div>
+    </div>
   </div>
+  {% endfor %}
+  {% endfor %}
+  {% endif %}
+</div>
 {% endblock %}
 
-
 {% block local_js_bottom %}
-  <script>
-  </script>
-  <script src="{% static 'custom/js/resize-ck.js' %}"></script>
-{% endblock %}
+<script src="{% static 'custom/js/resize-ck.js' %}"></script>
+<script>
+  $(document).ready(function () {
+    $('.manage-cohost').submit(function (e) {
+      e.preventDefault();
+      let $form = $(this);
+      $form.find('input[type="submit"]').prop('disabled', true);
+
+      let participant_id = e.target.participant.value;
+      let event_slug = e.target.event.value;
+      let submit_value = $form.find('input[type="submit"]').val();
+      let action = $form.find('input[type="submit"]').attr('data-action');
+      let csrftoken = e.target.csrfmiddlewaretoken.value;
+      let url = e.target.action;
+
+      $.ajax({
+        url: url,
+        type: 'POST',
+        data: {
+          'participant_id': participant_id,
+          'event_slug': event_slug,
+          'csrfmiddlewaretoken': csrftoken,
+          'action': action,
+          'submit': submit_value
+        },
+        success: function (data) {
+          $form.find('input[type="submit"]').prop('disabled', false);
+
+          console.log(data);
+
+          current_action = $form.find('input[type="submit"]').attr('data-action');
+          make_cohost_action = '{{ cohostStatus.MAKE_COHOST }}';
+          remove_cohost_action = '{{ cohostStatus.REMOVE_COHOST }}';
+
+          if (current_action == make_cohost_action) {
+            $form.find('input[type="submit"]').attr('data-action', remove_cohost_action);
+            $form.find('input[type="submit"]').val('Remove cohost');
+            $form.find('input[type="submit"]').removeClass('btn-primary');
+            $form.find('input[type="submit"]').addClass('btn-danger');
+          } else if (current_action == remove_cohost_action) {
+            $form.find('input[type="submit"]').attr('data-action', make_cohost_action);
+            $form.find('input[type="submit"]').val('Make cohost');
+            $form.find('input[type="submit"]').removeClass('btn-danger');
+            $form.find('input[type="submit"]').addClass('btn-success');
+          }
```

Comment:
Maybe we should split this into multiple functions. Right now it's a big chunk of code, handling success, error and all the independent logic could be function calls that are properly named. This would make the code more self-documenting